tensorboard/compat/tensorflow_stub/tensor_manip.py

```diff
@@ -0,0 +1,1005 @@
+    np.bool: SlowAppendBoolArrayToTensorProto,
+    dtypes.qint8.as_numpy_dtype: SlowAppendQIntArrayToTensorProto,
+    dtypes.quint8.as_numpy_dtype: SlowAppendQIntArrayToTensorProto,
+    dtypes.qint16.as_numpy_dtype: SlowAppendQIntArrayToTensorProto,
+    dtypes.quint16.as_numpy_dtype: SlowAppendQIntArrayToTensorProto,
+    dtypes.qint32.as_numpy_dtype: SlowAppendQIntArrayToTensorProto,
+    # NOTE(touts): Intentionally no way to feed a DT_BFLOAT16.
+}
+
+BACKUP_DICT = {dtypes.bfloat16.as_numpy_dtype: SlowAppendBFloat16ArrayToTensorProto}
+
+
+def GetFromNumpyDTypeDict(dtype_dict, dtype):
+    # NOTE: dtype_dict.get(dtype) always returns None.
+    for key, val in six.iteritems(dtype_dict):
+        if key == dtype:
+            return val
+    for key, val in six.iteritems(BACKUP_DICT):
+        if key == dtype:
+            return val
+    return None
+
+
+def GetNumpyAppendFn(dtype):
+    # numpy dtype for strings are variable length. We can not compare
+    # dtype with a single constant (np.string does not exist) to decide
+    # dtype is a "string" type. We need to compare the dtype.type to be
+    # sure it's a string type.
+    if dtype.type == np.string_ or dtype.type == np.unicode_:
+        # if _FAST_TENSOR_UTIL_AVAILABLE:
+        #   return fast_tensor_util.AppendObjectArrayToTensorProto
+        # else:
+        return SlowAppendObjectArrayToTensorProto
+    return GetFromNumpyDTypeDict(_NP_TO_APPEND_FN, dtype)
+
+
+# def TensorShapeProtoToList(shape):
+#   """Convert a TensorShape to a list.
+#
+#   Args:
+#     shape: A TensorShapeProto.
+#
+#   Returns:
+#     List of integers representing the dimensions of the tensor.
+#   """
+#   return [dim.size for dim in shape.dim]
+
+
+def _GetDenseDimensions(list_of_lists):
+    """Returns the inferred dense dimensions of a list of lists."""
+    if not isinstance(list_of_lists, (list, tuple)):
+        return []
+    elif not list_of_lists:
+        return [0]
+    else:
+        return [len(list_of_lists)] + _GetDenseDimensions(list_of_lists[0])
+
+
+def _FlattenToStrings(nested_strings):
+    if isinstance(nested_strings, (list, tuple)):
+        for inner in nested_strings:
+            for flattened_string in _FlattenToStrings(inner):
+                yield flattened_string
+    else:
+        yield nested_strings
+
+
+_TENSOR_CONTENT_TYPES = frozenset(
+    [
+        dtypes.float32,
+        dtypes.float64,
+        dtypes.int32,
+        dtypes.uint8,
+        dtypes.int16,
+        dtypes.int8,
+        dtypes.int64,
+        dtypes.qint8,
+        dtypes.quint8,
+        dtypes.qint16,
+        dtypes.quint16,
+        dtypes.qint32,
+        dtypes.uint32,
+        dtypes.uint64,
+    ]
+)
+
+
+class _Message(object):
+    def __init__(self, message):
+        self._message = message
+
+    def __repr__(self):
+        return self._message
+
+
+def _FirstNotNone(l):
+    for x in l:
+        if x is not None:
+            if isinstance(x, ops.Tensor):
```

Comment:
We can avoid the ops.Tensor reference here by just removing the special message in favor of:

```python
if x is not None:
    return x
```